controllers/dspipeline_controller.go

```diff
@@ -197,57 +200,203 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
-	usingCustomDB := params.UsingExternalDB(dspa)
+	// Initialize conditions
+	conditions := []metav1.Condition{}
 
+	usingCustomDB := params.UsingExternalDB(dspa)
+	databaseReady := r.buildCondition("DatabaseReady", dspa.Generation, "DatabaseReady")
 	if !usingCustomDB {
+
 		err = r.ReconcileDatabase(ctx, dspa, params)
 		if err != nil {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
 			return ctrl.Result{}, err
 		}
+
+		if !r.isDeploymentAvailable(ctx, dspa, "mariadb") {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
+
+			err = r.Status().Update(ctx, dspa)
```

Comment:
because you are using: `condition.LastTransitionTime = metav1.Now()` in `buildCondition` we will always update the status field every reconcilliation loop, which will trigger new reconcilliations every update, even if the status/reasons don't change, we want to avoid this and only update if a change occurs. 

which makes me wonder if this is all possible without putting watches on deployment status changes, you might have to do this once you put that condition in